auth_user_basic.go

```diff
@@ -57,6 +57,26 @@ func (u *basicUser) HasPermission(opts PermissionOpts) (bool, error) {
 	}
 	return false, nil
 }
+func (u *basicUser) PermissionsForResource(opts PermissionOpts) (map[string]int, error) {
```

Comment:
I would be inclined to make this a public function that takes User, PermissionOpts, RoleManager (or make it a RoleManager method? but the truth is that it's not a thing that is going to change based on implementation and doesn't really need to have anything inside of the user...

Reply:
changed to a public function